test/integration/testpaths/testpath_snapshot_limits.py

```diff
@@ -0,0 +1,355 @@
+
+        self.assertNotEqual(len(snapshots_list),
+                            0,
+                            "Check listSnapshots response")
+
+        # Verify Snapshot state
+        self.assertEqual(
+            snapshots_list[0].state in [
+                'BackedUp',
+            ],
+            True,
+            "Snapshot state is not as expected. It is %s" %
+            snapshots_list[0].state
+        )
+
+        # Step 2
+        qryresult_after_snapshot = self.dbclient.execute(
+            " select id, account_name, secondaryStorageTotal\
+                        from account_view where account_name = '%s';" %
+            self.account.name)
+
+        self.assertNotEqual(
+            len(qryresult_after_snapshot),
+            0,
+            "Check sql query to return SecondaryStorageTotal of account")
+
+        secStorageAfterSnapshotCreated = qryresult_after_snapshot[0][2]
+
+        snapshot_size = snapshots_list[0].physicalsize
+        secStorageIncreased = secStorageBeforeSnapshot + \
+            snapshot_size
+
+        self.assertEqual(
+            secStorageIncreased,
+            secStorageAfterSnapshotCreated,
+            "Secondary storage Total after Snapshot\
+                        should be incremented by size of snapshot.")
+
+        # Step 3
+        snapshot.delete(self.apiclient)
+
+        snapshots_list = Snapshot.list(self.userapiclient,
+                                       id=snapshot.id)
+
+        self.assertEqual(
+            snapshots_list,
+            None,
+            "Snapshot not deleted."
+        )
+
+        # Step 4
+        qryresult_after_snapshot_deleted = self.dbclient.execute(
+            " select id, account_name, secondaryStorageTotal\
+                        from account_view where account_name = '%s';" %
+            self.account.name)
+
+        self.assertNotEqual(
+            len(qryresult_after_snapshot_deleted),
+            0,
+            "Check sql query to return SecondaryStorageTotal of account")
+
+        secStorageAfterSnapshotDeleted = qryresult_after_snapshot_deleted[0][2]
+
+        secStorageDecreased = secStorageAfterSnapshotCreated - \
+            snapshot_size
+
+        self.assertEqual(
+            secStorageDecreased,
+            secStorageAfterSnapshotDeleted,
+            "Secondary storage Total after Snapshot\
+                        should be incremented by size of snapshot.")
+
+        # Step 5
+        # Set Snapshot Limit for account
+        Resources.updateLimit(self.apiclient, resourcetype=3,
+                              max=1, account=self.account.name,
+                              domainid=self.account.domainid)
+
+        # Step 6
+        snapshot = Snapshot.create(
+            self.userapiclient,
+            root_volume.id)
+
+        snapshots_list = Snapshot.list(self.userapiclient,
+                                       id=snapshot.id)
+
+        self.assertEqual(isinstance(snapshots_list, list),
```

Comment:
Avoid using isinstance and len methods. Use validateList method from utils instead